main/src/components/docs/LegendItemNew.jsx

```diff
@@ -0,0 +1,234 @@
+import { useEffect, useState } from 'react';
+import styles from './LegendItem.module.css';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+
+export default function LegendItemNew({ svgPath, itemsName }) {
+
+  const [splitSvgsDay, setSplitSvgsDay] = useState([]);
+  const [splitSvgsNight, setSplitSvgsNight] = useState([]);
+  const [error, setError] = useState(null);
+  const [svgContentDay, setSvgContentDay] = useState('');
+  const [svgContentNight, setSvgContentNight] = useState('');
+  const [loadingDay, setLoadingDay] = useState(true);
+  const [loadingNight, setLoadingNight] = useState(true);
+
+  useEffect(() => {
+    if (!svgPath) {
+      setSvgContentDay('');
+      setSvgContentNight('');
+      setLoadingDay(false);
+      setLoadingNight(false);
+      return;
+    }
+    setLoadingDay(true);
+    setLoadingNight(true);
+    setError(null);
+
+    fetch(svgPath + "-day.svg")
+      .then(response => {
+        if (!response.ok) {
+          throw new Error(`HTTP error! status: ${response.status}`);
+        }
+        return response.text();
+      })
+      .then(data => {
+        setSvgContentDay(data);
+        setLoadingDay(false);
+      })
+      .catch(err => {
+        console.error('Error fetching SVG:', err);
+        setError(err);
+        setLoadingDay(false);
+      });
+    fetch(svgPath + "-day.svg")
+      .then(response => {
+        if (!response.ok) {
+          throw new Error(`HTTP error! status: ${response.status}`);
+        }
+        return response.text();
+      })
+      .then(data => {
+        setSvgContentNight(data);
+        setLoadingNight(false);
+      })
+      .catch(err => {
+        console.error('Error fetching SVG:', err);
+        setError(err);
+        setLoadingNight(false);
+      });
+  }, [svgPath]);
+
+  function splitSvgToArray(svgContent) {
```

Comment:
Extract a common split function splitSvgContent and reuse it for both day and night